Orange/data/filter.py

```diff
@@ -70,6 +76,12 @@ def __call__(self, data):
             r = np.logical_not(r)
         return data[r]
 
+    def __eq__(self, other):
+        return super().__eq__(other) and self.columns == other.columns
+
+    def __hash__(self):
+        return super().__hash__() ^ hash(self.columns)
```

Comment:
Why xor and not putting it within one `hash()`?

Reply:
It would actually be much better to use a hash of a tuple. Tuple's hash is much smarter than xor: https://github.com/python/cpython/blob/main/Objects/tupleobject.c#L289

```
>>> a = "Ana"
>>> b = "Berta"
>>> hash(a) ^ hash(a)
0
>>> hash(b) ^ hash(b)
0
```

Obviously.

Whereas:

```
>>> hash((a, a))
3546956001028672602
>>> hash((b, b))
6385707759775101969
```

Reply:
I know it is probably not significant, but I changed it anyway.